frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.unit.spec.tsx

```diff
@@ -0,0 +1,135 @@
+      }),
+    }),
+  });
+};
+
+describe("DashboardEmbedHeaderButton", () => {
+  describe("whether the button should be rendered", () => {
+    describe("when the user is admin", () => {
+      it("should render the button even if there is no public link", () => {
+        setup({ isAdmin: true, hasPublicLink: false });
+        expect(screen.getByLabelText("share icon")).toBeInTheDocument();
+      });
+    });
+
+    describe("when the user is non-admin", () => {
+      it("should not render the button if there is no public link", () => {
+        setup({
+          hasPublicLink: false,
+          publicLinksEnabled: false,
+          isAdmin: false,
+        });
+        expect(screen.queryByLabelText("share icon")).not.toBeInTheDocument();
+      });
+
+      it("should not render the button if public sharing is enabled but no public link", () => {
+        setup({
+          hasPublicLink: false,
+          isAdmin: false,
+          publicLinksEnabled: true,
+        });
+        expect(screen.queryByLabelText("share icon")).not.toBeInTheDocument();
+      });
+
+      // TODO: Test for rendering button for non-admins when a public link is created
+    });
+  });
+
+  describe("when the popover or modal should be rendered", () => {
+    it("should render the popover when public sharing is true", async () => {
+      setup({ publicLinksEnabled: true, isAdmin: true });
+      userEvent.click(screen.getByLabelText("share icon"));
+      expect(
+        await screen.findByTestId("embed-header-menu"),
+      ).toBeInTheDocument();
+    });
+
+    it("should render the embedding modal when public sharing is false", async () => {
+      setup({ publicLinksEnabled: false, isAdmin: true });
+      userEvent.click(await screen.findByLabelText("share icon"));
+
+      expect(
+        await screen.findByTestId("dashboard-sharing-embedding-modal"),
+      ).toBeInTheDocument();
+    });
+  });
+
+  describe("which popover content should be rendered", () => {
+    describe("when user is admin", () => {
+      it("should render the menu with `Create a public link` if a link hasn't been created", async () => {
+        setup({
+          hasPublicLink: false,
+          publicLinksEnabled: true,
+          isAdmin: true,
+        });
+        userEvent.click(screen.getByLabelText("share icon"));
+        expect(
+          await screen.findByText("Create a public link"),
+        ).toBeInTheDocument();
+        expect(await screen.findByText("Embed")).toBeInTheDocument();
+      });
+
+      it("should render the menu with `Public link` option if a public link has been created", async () => {
+        setup({
+          hasPublicLink: true,
+          publicLinksEnabled: true,
+          isAdmin: true,
+        });
+
+        userEvent.click(screen.getByLabelText("share icon"));
+
+        expect(await screen.findByText("Embed")).toBeInTheDocument();
+        expect(await screen.findByText("Public link")).toBeInTheDocument();
```

Comment:
nit: Moving this above `Embed` would make the assertion order the same as the test before.